web/src/message_feed_top_notices.ts

```diff
@@ -31,9 +30,10 @@ function show_end_of_results_notice(): void {
     // Note that element we adjust is not visible to spectators.
     const narrow_filter = narrow_state.filter();
     assert(narrow_filter !== undefined);
-    const terms = narrow_filter.terms();
-    const update_hash = hash_util.search_public_streams_notice_url(terms);
-    $(".all-messages-search-caution a.search-shared-history").attr("href", update_hash);
+    $(".all-messages-search-caution a.search-shared-history").attr(
+        "href",
+        "help/search-for-messages#search-shared-history",
+    );
```

Comment:
There's no need to set this here instead of the template. You do need to update this function for the removal of the shared history link in the template.